doc/main.doc.md

```diff
@@ -64,27 +64,53 @@ Each module `xxx` has a `xxx.init.s` file with its initialization code.
 
 ## Main Loop
 
-Do nothing, we work in `VBlank`.
+The main logic in included in `body.s`.
+
+We use a look to synchronize the update of the values with the update of the screen. After each iteration we wait for a VBlank, and if we didn't finished the iteration, the VBlank interruption do nothing.
 ~~~
 ; ///////// MAIN LOOP \\\\\\\\\
 loop:
-	jr loop
+; //// WAIT FOR VBLANK \\\\
+	halt
+	ld a,(lock)
+	and a
+  jp nz,loop			; wait until lock = 0
+; \\\\ WAIT FOR VBLANK ////
+
+.INCLUDE "body.s"
+
+; //// ALLOW WBLANK TO UPDATE THE SCREEN \\\\
+	ld a,1
+	ld (lock),a    ; lock = 1
+; \\\\ ALLOW WBLANK TO UPDATE THE SCREEN ////
+; \\\\\\\\\ MAIN LOOP /////////
+	jp loop
 ; \\\\\\\\\ MAIN LOOP /////////
 ~~~
 
 ## VBlank
 
-The VBLank function. **We include the `display` module first.**. The VLblank is a save time to edit the screen, and after this time, we can't edit the screen anymore, but we can still compute the values for the next frame.
-
-If the module `xxx` (`display` aside) needs to execute code at each iteration, the file `xxx.s` is included inside the file `body.s`.
+The VBlank function. This function is called with VBlank interrupt, but we only update the screen after each iteration of the loop. So, when the lock is not in the right state, we return at once. Else we run the code include in `xxx.vbl.s` files (the display module first, since we can only update the screen during the vblank time). The code included here has to be really fast. Long computations should be done in the loop.
 
 ~~~
 ; ///////// VBlank Interuption \\\\\\\\\
 VBlank:
 	push af
 	push hl

```

Comment:
In an interrupt, all registers are callee saved !  
You don't save bc nor de, and they might be modified in the VBlank interrupt.

Reply:
They are not supposed to be modified except between the loop iterations, but yes, better be safe than sorry.